Add i18n guidance re. images & shortcodes + some rework (#4790)
open-telemetry/opentelemetry.io: content/en/docs/contributing/localization.md

---
title: Site localization
description: Creating and maintaining site pages in non-English localizations.
linkTitle: Localization
weight: 25
cSpell:ignore: shortcodes
---

The OTel website uses Hugo's [multilingual framework] to support page localizations.
English is the default language, with US English as the default (implicit) localization.
A growing number of other localizations are supported, as can be seen from the languages
dropdown menu in the top nav.

## Translation guidance

We recommend that you follow the guidance offered in this section when
translating pages.

### Heading anchors

To ensure that heading anchor targets are uniform across localizations, when
translating headings:

- Preserve the heading's explicit ID if it has one. [Heading ID syntax] is

>   written after the heading text using syntax like `{ #some-id }`

Reviewer: ```suggestion
  written after the heading text using syntax like `{ #some-id }`.
```

- Otherwise, explicitly declare a heading ID corresponding to the autogenerated
  ID of the original English heading.

{{% alert title="Note" %}}

We leave it to the discretion of localization authors to decide if they add an
explicit heading ID to all headings of translated pages, or whether this is only

> done for known intra-site heading targets -- as reported by the link checker.

Reviewer: ```suggestion
done for known intra-site heading targets, as reported by the link checker.
```

> The former is more uniform and more work, but better supports having
> site-external targets into localization pages, and avoids having to revisit
> previously translated pages as new heading targets are used.

Reviewer: ```suggestion
The former option is more uniform and more work. It better supports having
site-external targets into localization pages and avoids having to revisit
previously translated pages as new heading targets are used.
```

{{% /alert %}}

[Heading ID syntax]:
  https://github.com/yuin/goldmark/blob/master/README.md#headings

### Links

For link references to local paths (as opposed to external links), preserve the
path _as is_. This holds true for links to website pages or section-local
resources such as images.

{{% alert title="Note" %}}

> The OTel website repository has a custom render-link hook that Hugo uses to
> convert **absolute link paths to documentation pages** that are of the form
> `/docs/some-page`, to be locale specific, by prefixing the path with page
> language code when rendering the link. For example, the previous sample path

Reviewer: ```suggestion
The OTel website repository has a custom render-link hook that Hugo uses to
convert **absolute link paths to documentation pages**. Links of the form
`/docs/some-page` are made locale specific by prefixing the path with the page
language code when rendering the link. For example, the previous sample path
```

would become `/ja/docs/some-page` when rendered from a Japanese page.

{{% /alert %}}

### Images

Hugo is smart in the way that it renders page images that are shared across site
localizations. That is, in the generated site folder, Hugo will output a
_single_ image file that is then shared across localizations.

So as a general rule, _do not_ make copies of image files in your localization
unless you actually change the image.

### Shortcodes

Some of the base shortcodes contain English text that you might need to localize

> -- this is particularly true of those contain in [layouts/shortcodes/docs].

Reviewer: ```suggestion
-- this is particularly true of those contained in [layouts/shortcodes/docs].
```
